traffic_ops/app/lib/API/Server.pm

```diff
@@ -911,6 +911,21 @@ sub is_server_valid {
 		return ( 0, "Invalid server type" );
 	}
 
+	my $cdn_mismatch;
+	if ($id) {
+		my $profile = $self->db->resultset('Profile')->search( { 'me.id' => $params->{profileId}}, { prefetch => ['cdn'] } )->single();
+		if ( !defined($profile->cdn) ) {
+			$cdn_mismatch = 1;
+		} 
+		elsif ( $params->{cdnId} != $profile->cdn->id ) {
+			$cdn_mismatch = 1;
+		}
+	}
+
+	if ($cdn_mismatch) {
+		return ( 0, "CDN of profile does not match Server CDN" );
+	}
```

Comment:
you could simplify all this code to simply:

if ($id) {
		my $profile = $self->db->resultset('Profile')->search( { 'me.id' => $params->{profileId}}, { prefetch => ['cdn'] } )->single();
		if ( !defined($profile->cdn) || $params->{cdnId} != $profile->cdn->id ) {
			return ( 0, "CDN of profile does not match Server CDN" );
		}
	}